%% Generated by Sphinx.
\def\sphinxdocclass{report}
\documentclass[letterpaper,10pt,english]{sphinxmanual}
\ifdefined\pdfpxdimen
   \let\sphinxpxdimen\pdfpxdimen\else\newdimen\sphinxpxdimen
\fi \sphinxpxdimen=.75bp\relax
\ifdefined\pdfimageresolution
    \pdfimageresolution= \numexpr \dimexpr1in\relax/\sphinxpxdimen\relax
\fi
%% let collapsible pdf bookmarks panel have high depth per default
\PassOptionsToPackage{bookmarksdepth=5}{hyperref}

\PassOptionsToPackage{booktabs}{sphinx}
\PassOptionsToPackage{colorrows}{sphinx}

\PassOptionsToPackage{warn}{textcomp}
\usepackage[utf8]{inputenc}
\ifdefined\DeclareUnicodeCharacter
% support both utf8 and utf8x syntaxes
  \ifdefined\DeclareUnicodeCharacterAsOptional
    \def\sphinxDUC#1{\DeclareUnicodeCharacter{"#1}}
  \else
    \let\sphinxDUC\DeclareUnicodeCharacter
  \fi
  \sphinxDUC{00A0}{\nobreakspace}
  \sphinxDUC{2500}{\sphinxunichar{2500}}
  \sphinxDUC{2502}{\sphinxunichar{2502}}
  \sphinxDUC{2514}{\sphinxunichar{2514}}
  \sphinxDUC{251C}{\sphinxunichar{251C}}
  \sphinxDUC{2572}{\textbackslash}
\fi
\usepackage{cmap}
\usepackage[T1]{fontenc}
\usepackage{amsmath,amssymb,amstext}
\usepackage{babel}



\usepackage{tgtermes}
\usepackage{tgheros}
\renewcommand{\ttdefault}{txtt}



\usepackage[Bjarne]{fncychap}
\usepackage{sphinx}

\fvset{fontsize=auto}
\usepackage{geometry}


% Include hyperref last.
\usepackage{hyperref}
% Fix anchor placement for figures with captions.
\usepackage{hypcap}% it must be loaded after hyperref.
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}

\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\usepackage{sphinxmessages}
\setcounter{tocdepth}{1}



\title{Trajectome}
\date{Feb 13, 2025}
\release{1.0}
\author{Neelesh Soni}
\newcommand{\sphinxlogo}{\vbox{}}
\renewcommand{\releasename}{Release}
\makeindex
\begin{document}

\ifdefined\shorthandoff
  \ifnum\catcode`\=\string=\active\shorthandoff{=}\fi
  \ifnum\catcode`\"=\active\shorthandoff{"}\fi
\fi

\pagestyle{empty}
\sphinxmaketitle
\pagestyle{plain}
\sphinxtableofcontents
\pagestyle{normal}
\phantomsection\label{\detokenize{index::doc}}


\sphinxstepscope


\chapter{Trajectome}
\label{\detokenize{modules:trajectome}}\label{\detokenize{modules::doc}}
\sphinxstepscope


\section{Trajectome}
\label{\detokenize{src:trajectome}}\label{\detokenize{src::doc}}

\subsection{Submodules}
\label{\detokenize{src:submodules}}

\subsection{src.Interaction\_Class module}
\label{\detokenize{src:module-src.Interaction_Class}}\label{\detokenize{src:src-interaction-class-module}}\index{module@\spxentry{module}!src.Interaction\_Class@\spxentry{src.Interaction\_Class}}\index{src.Interaction\_Class@\spxentry{src.Interaction\_Class}!module@\spxentry{module}}
\sphinxAtStartPar
Author: Neelesh Soni, \sphinxhref{mailto:neelesh@salilab.org}{neelesh@salilab.org}, \sphinxhref{mailto:neeleshsoni03@gmail.com}{neeleshsoni03@gmail.com}
Date: April 5, 2024
\index{logger (in module src.Interaction\_Class)@\spxentry{logger}\spxextra{in module src.Interaction\_Class}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Interaction_Class.logger}}
\pysigstartsignatures
\pysigline{\sphinxcode{\sphinxupquote{src.Interaction\_Class.}}\sphinxbfcode{\sphinxupquote{logger}}}
\pysigstopsignatures
\sphinxAtStartPar
Description
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
TYPE

\end{description}\end{quote}

\end{fulllineitems}

\index{Interaction (class in src.Interaction\_Class)@\spxentry{Interaction}\spxextra{class in src.Interaction\_Class}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Interaction_Class.Interaction}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{ }}}\sphinxcode{\sphinxupquote{src.Interaction\_Class.}}\sphinxbfcode{\sphinxupquote{Interaction}}}{\sphinxparam{\DUrole{n}{system}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
A class to represent interactions between proteins within a simulation system.

\sphinxAtStartPar
The \sphinxtitleref{Interaction} class is responsible for defining and adding various types of
restraints between proteins to model their interactions in a Brownian dynamics
simulation.
\index{system (src.Interaction\_Class.Interaction attribute)@\spxentry{system}\spxextra{src.Interaction\_Class.Interaction attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Interaction_Class.Interaction.system}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{system}}}
\pysigstopsignatures
\sphinxAtStartPar
The simulation system in which the interactions are defined.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
{\hyperref[\detokenize{src:src.System_Class.System}]{\sphinxcrossref{System}}}

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_binding\_restraint() (src.Interaction\_Class.Interaction method)@\spxentry{add\_binding\_restraint()}\spxextra{src.Interaction\_Class.Interaction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Interaction_Class.Interaction.add_binding_restraint}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_binding\_restraint}}}{\sphinxparam{\DUrole{n}{prot1\_tuple}}\sphinxparamcomma \sphinxparam{\DUrole{n}{prot2\_tuple}}\sphinxparamcomma \sphinxparam{\DUrole{n}{name}}\sphinxparamcomma \sphinxparam{\DUrole{n}{mean\_dist}}\sphinxparamcomma \sphinxparam{\DUrole{n}{kappa}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Adds a harmonic binding restraint between two proteins.

\sphinxAtStartPar
This function defines a distance\sphinxhyphen{}based harmonic restraint between
two protein particles, ensuring they remain within a defined mean
distance with a given kappa value.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{prot1\_tuple}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} A tuple containing the first protein name and its index.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{prot2\_tuple}} (\sphinxstyleliteralemphasis{\sphinxupquote{tuple}}) \textendash{} A tuple containing the second protein name and its index.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The name of the restraint.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{mean\_dist}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} The expected mean distance between the proteins.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{kappa}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} The force constant for the harmonic function.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_distance\_restraint() (src.Interaction\_Class.Interaction method)@\spxentry{add\_distance\_restraint()}\spxextra{src.Interaction\_Class.Interaction method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Interaction_Class.Interaction.add_distance_restraint}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_distance\_restraint}}}{\sphinxparam{\DUrole{n}{prot1}}\sphinxparamcomma \sphinxparam{\DUrole{n}{prot2}}\sphinxparamcomma \sphinxparam{\DUrole{n}{dist}}\sphinxparamcomma \sphinxparam{\DUrole{n}{k}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Adds a distance\sphinxhyphen{}based restraint between two proteins.

\sphinxAtStartPar
This function creates a distance restraint to enforce a specific separation
between two proteins. The restraint applies a force that maintains the
proteins at a given distance with the specified force constant.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{prot1}} ({\hyperref[\detokenize{src:src.Protein_Class.Protein}]{\sphinxcrossref{\sphinxstyleliteralemphasis{\sphinxupquote{Protein}}}}}) \textendash{} The first protein.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{prot2}} ({\hyperref[\detokenize{src:src.Protein_Class.Protein}]{\sphinxcrossref{\sphinxstyleliteralemphasis{\sphinxupquote{Protein}}}}}) \textendash{} The second protein.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{dist}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} The target distance for the restraint.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{k}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} The force constant for the restraint.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}



\subsection{src.Protein\_Class module}
\label{\detokenize{src:module-src.Protein_Class}}\label{\detokenize{src:src-protein-class-module}}\index{module@\spxentry{module}!src.Protein\_Class@\spxentry{src.Protein\_Class}}\index{src.Protein\_Class@\spxentry{src.Protein\_Class}!module@\spxentry{module}}
\sphinxAtStartPar
Author: Neelesh Soni, \sphinxhref{mailto:neelesh@salilab.org}{neelesh@salilab.org}, \sphinxhref{mailto:neeleshsoni03@gmail.com}{neeleshsoni03@gmail.com}
Date: April 5, 2024

\sphinxAtStartPar
This module defines the \sphinxtitleref{Protein} class used for protein simulations.
\index{logger (in module src.Protein\_Class)@\spxentry{logger}\spxextra{in module src.Protein\_Class}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.logger}}
\pysigstartsignatures
\pysigline{\sphinxcode{\sphinxupquote{src.Protein\_Class.}}\sphinxbfcode{\sphinxupquote{logger}}}
\pysigstopsignatures
\sphinxAtStartPar
Logger for the module.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
logging.Logger

\end{description}\end{quote}

\end{fulllineitems}

\index{Protein (class in src.Protein\_Class)@\spxentry{Protein}\spxextra{class in src.Protein\_Class}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{ }}}\sphinxcode{\sphinxupquote{src.Protein\_Class.}}\sphinxbfcode{\sphinxupquote{Protein}}}{\sphinxparam{\DUrole{n}{model}}\sphinxparamcomma \sphinxparam{\DUrole{n}{state}}\sphinxparamcomma \sphinxparam{\DUrole{n}{h\_root}}\sphinxparamcomma \sphinxparam{\DUrole{n}{name}}\sphinxparamcomma \sphinxparam{\DUrole{n}{fastafile}}\sphinxparamcomma \sphinxparam{\DUrole{n}{prot\_param}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
A class to represent a protein in a simulation.
\index{model (src.Protein\_Class.Protein attribute)@\spxentry{model}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.model}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{model}}}
\pysigstopsignatures
\sphinxAtStartPar
The IMP model.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.Model

\end{description}\end{quote}

\end{fulllineitems}

\index{name (src.Protein\_Class.Protein attribute)@\spxentry{name}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.name}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{name}}}
\pysigstopsignatures
\sphinxAtStartPar
Name of the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
str

\end{description}\end{quote}

\end{fulllineitems}

\index{center (src.Protein\_Class.Protein attribute)@\spxentry{center}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.center}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{center}}}
\pysigstopsignatures
\sphinxAtStartPar
Coordinates of the protein’s center.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{radius (src.Protein\_Class.Protein attribute)@\spxentry{radius}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.radius}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{radius}}}
\pysigstopsignatures
\sphinxAtStartPar
Radius of the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
int

\end{description}\end{quote}

\end{fulllineitems}

\index{mass (src.Protein\_Class.Protein attribute)@\spxentry{mass}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.mass}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{mass}}}
\pysigstopsignatures
\sphinxAtStartPar
Mass of the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
int

\end{description}\end{quote}

\end{fulllineitems}

\index{diffcoff (src.Protein\_Class.Protein attribute)@\spxentry{diffcoff}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.diffcoff}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{diffcoff}}}
\pysigstopsignatures
\sphinxAtStartPar
Diffusion coefficient of the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{color (src.Protein\_Class.Protein attribute)@\spxentry{color}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.color}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{color}}}
\pysigstopsignatures
\sphinxAtStartPar
Color index for display.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
int

\end{description}\end{quote}

\end{fulllineitems}

\index{protp (src.Protein\_Class.Protein attribute)@\spxentry{protp}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.protp}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{protp}}}
\pysigstopsignatures
\sphinxAtStartPar
The main particle representing the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.Particle

\end{description}\end{quote}

\end{fulllineitems}

\index{hier (src.Protein\_Class.Protein attribute)@\spxentry{hier}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.hier}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{hier}}}
\pysigstopsignatures
\sphinxAtStartPar
Hierarchy setup for the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.atom.Hierarchy

\end{description}\end{quote}

\end{fulllineitems}

\index{prb (src.Protein\_Class.Protein attribute)@\spxentry{prb}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.prb}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{prb}}}
\pysigstopsignatures
\sphinxAtStartPar
Rigid body of the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.core.RigidBody

\end{description}\end{quote}

\end{fulllineitems}

\index{mol (src.Protein\_Class.Protein attribute)@\spxentry{mol}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.mol}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{mol}}}
\pysigstopsignatures
\sphinxAtStartPar
Molecule representation of the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.atom.Molecule

\end{description}\end{quote}

\end{fulllineitems}

\index{dif (src.Protein\_Class.Protein attribute)@\spxentry{dif}\spxextra{src.Protein\_Class.Protein attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.dif}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{dif}}}
\pysigstopsignatures
\sphinxAtStartPar
Diffusion setup for the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.atom.Diffusion

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_rigid\_body\_protein() (src.Protein\_Class.Protein method)@\spxentry{create\_rigid\_body\_protein()}\spxextra{src.Protein\_Class.Protein method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.create_rigid_body_protein}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{create\_rigid\_body\_protein}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_current\_localization() (src.Protein\_Class.Protein method)@\spxentry{get\_current\_localization()}\spxextra{src.Protein\_Class.Protein method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.get_current_localization}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_current\_localization}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{get\_radius\_mass\_from\_fastafile() (src.Protein\_Class.Protein method)@\spxentry{get\_radius\_mass\_from\_fastafile()}\spxextra{src.Protein\_Class.Protein method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.Protein.get_radius_mass_from_fastafile}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_radius\_mass\_from\_fastafile}}}{}{}
\pysigstopsignatures
\end{fulllineitems}


\end{fulllineitems}

\index{ProteinStructure (class in src.Protein\_Class)@\spxentry{ProteinStructure}\spxextra{class in src.Protein\_Class}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{ }}}\sphinxcode{\sphinxupquote{src.Protein\_Class.}}\sphinxbfcode{\sphinxupquote{ProteinStructure}}}{\sphinxparam{\DUrole{n}{model}}\sphinxparamcomma \sphinxparam{\DUrole{n}{state}}\sphinxparamcomma \sphinxparam{\DUrole{n}{h\_root}}\sphinxparamcomma \sphinxparam{\DUrole{n}{name}}\sphinxparamcomma \sphinxparam{\DUrole{n}{pdbfile}}\sphinxparamcomma \sphinxparam{\DUrole{n}{prot\_param}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
A class representing a protein structure in the simulation.

\sphinxAtStartPar
The \sphinxtitleref{ProteinStructure} class loads protein structural information from PDB or CIF files,
processes it into a hierarchical representation, and applies coarse\sphinxhyphen{}grained modeling
for molecular simulations. It also supports setting up diffusion properties and
rigid\sphinxhyphen{}body representations.
\index{model (src.Protein\_Class.ProteinStructure attribute)@\spxentry{model}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.model}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{model}}}
\pysigstopsignatures
\sphinxAtStartPar
The IMP model used for simulation.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.Model

\end{description}\end{quote}

\end{fulllineitems}

\index{state (src.Protein\_Class.ProteinStructure attribute)@\spxentry{state}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.state}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{state}}}
\pysigstopsignatures
\sphinxAtStartPar
The state of the protein in the simulation.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.pmi.topology.State

\end{description}\end{quote}

\end{fulllineitems}

\index{h\_root (src.Protein\_Class.ProteinStructure attribute)@\spxentry{h\_root}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.h_root}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{h\_root}}}
\pysigstopsignatures
\sphinxAtStartPar
Root hierarchy of the system.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.atom.Hierarchy

\end{description}\end{quote}

\end{fulllineitems}

\index{name (src.Protein\_Class.ProteinStructure attribute)@\spxentry{name}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.name}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{name}}}
\pysigstopsignatures
\sphinxAtStartPar
The name of the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
str

\end{description}\end{quote}

\end{fulllineitems}

\index{pdbfile (src.Protein\_Class.ProteinStructure attribute)@\spxentry{pdbfile}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.pdbfile}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{pdbfile}}}
\pysigstopsignatures
\sphinxAtStartPar
Path to the PDB file containing the protein structure.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
str

\end{description}\end{quote}

\end{fulllineitems}

\index{multi\_model (src.Protein\_Class.ProteinStructure attribute)@\spxentry{multi\_model}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.multi_model}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{multi\_model}}}
\pysigstopsignatures
\sphinxAtStartPar
Whether the PDB file contains multiple models.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{resolution (src.Protein\_Class.ProteinStructure attribute)@\spxentry{resolution}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.resolution}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{resolution}}}
\pysigstopsignatures
\sphinxAtStartPar
Resolution level for coarse\sphinxhyphen{}graining the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
int

\end{description}\end{quote}

\end{fulllineitems}

\index{color (src.Protein\_Class.ProteinStructure attribute)@\spxentry{color}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.color}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{color}}}
\pysigstopsignatures
\sphinxAtStartPar
Color index for visualization.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
int

\end{description}\end{quote}

\end{fulllineitems}

\index{centerize (src.Protein\_Class.ProteinStructure attribute)@\spxentry{centerize}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.centerize}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{centerize}}}
\pysigstopsignatures
\sphinxAtStartPar
Whether to centerize the protein coordinates.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
bool

\end{description}\end{quote}

\end{fulllineitems}

\index{diffcoff (src.Protein\_Class.ProteinStructure attribute)@\spxentry{diffcoff}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.diffcoff}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{diffcoff}}}
\pysigstopsignatures
\sphinxAtStartPar
Translational diffusion coefficient of the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{rot\_diffcoff\_scale (src.Protein\_Class.ProteinStructure attribute)@\spxentry{rot\_diffcoff\_scale}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.rot_diffcoff_scale}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{rot\_diffcoff\_scale}}}
\pysigstopsignatures
\sphinxAtStartPar
Scaling factor for rotational diffusion.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{protein (src.Protein\_Class.ProteinStructure attribute)@\spxentry{protein}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.protein}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{protein}}}
\pysigstopsignatures
\sphinxAtStartPar
IMP particle representing the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.Particle

\end{description}\end{quote}

\end{fulllineitems}

\index{hier (src.Protein\_Class.ProteinStructure attribute)@\spxentry{hier}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.hier}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{hier}}}
\pysigstopsignatures
\sphinxAtStartPar
Hierarchical representation of the protein structure.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.atom.Hierarchy

\end{description}\end{quote}

\end{fulllineitems}

\index{mol (src.Protein\_Class.ProteinStructure attribute)@\spxentry{mol}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.mol}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{mol}}}
\pysigstopsignatures
\sphinxAtStartPar
IMP molecule representation for managing residues.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.atom.Molecule

\end{description}\end{quote}

\end{fulllineitems}

\index{prb (src.Protein\_Class.ProteinStructure attribute)@\spxentry{prb}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.prb}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{prb}}}
\pysigstopsignatures
\sphinxAtStartPar
Rigid body representation of the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.core.RigidBody

\end{description}\end{quote}

\end{fulllineitems}

\index{amino\_acid\_radii (src.Protein\_Class.ProteinStructure attribute)@\spxentry{amino\_acid\_radii}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.amino_acid_radii}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{amino\_acid\_radii}}}
\pysigstopsignatures
\sphinxAtStartPar
Dictionary mapping amino acid names to their default radii.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
dict

\end{description}\end{quote}

\end{fulllineitems}

\index{amino\_acid\_mass (src.Protein\_Class.ProteinStructure attribute)@\spxentry{amino\_acid\_mass}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.amino_acid_mass}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{amino\_acid\_mass}}}
\pysigstopsignatures
\sphinxAtStartPar
Dictionary mapping amino acid names to their default masses.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
dict

\end{description}\end{quote}

\end{fulllineitems}

\index{All\_Residues (src.Protein\_Class.ProteinStructure attribute)@\spxentry{All\_Residues}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.All_Residues}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{All\_Residues}}}
\pysigstopsignatures
\sphinxAtStartPar
Ordered dictionary storing protein residues.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
OrderedDict

\end{description}\end{quote}

\end{fulllineitems}

\index{Coarse\_Residues (src.Protein\_Class.ProteinStructure attribute)@\spxentry{Coarse\_Residues}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.Coarse_Residues}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{Coarse\_Residues}}}
\pysigstopsignatures
\sphinxAtStartPar
List of coarse\sphinxhyphen{}grained residue fragments.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{Protein\_Residues (src.Protein\_Class.ProteinStructure attribute)@\spxentry{Protein\_Residues}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.Protein_Residues}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{Protein\_Residues}}}
\pysigstopsignatures
\sphinxAtStartPar
List of IMP particles representing residues.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{Tran\_dif (src.Protein\_Class.ProteinStructure attribute)@\spxentry{Tran\_dif}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.Tran_dif}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{Tran\_dif}}}
\pysigstopsignatures
\sphinxAtStartPar
Translational diffusion object for the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.atom.Diffusion

\end{description}\end{quote}

\end{fulllineitems}

\index{Rot\_diff (src.Protein\_Class.ProteinStructure attribute)@\spxentry{Rot\_diff}\spxextra{src.Protein\_Class.ProteinStructure attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.Rot_diff}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{Rot\_diff}}}
\pysigstopsignatures
\sphinxAtStartPar
Rotational diffusion object for the protein.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.atom.RigidBodyDiffusion

\end{description}\end{quote}

\end{fulllineitems}

\index{Get\_Hierarchy\_From\_PDB() (src.Protein\_Class.ProteinStructure method)@\spxentry{Get\_Hierarchy\_From\_PDB()}\spxextra{src.Protein\_Class.ProteinStructure method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.Get_Hierarchy_From_PDB}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{Get\_Hierarchy\_From\_PDB}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Extracts the hierarchy of the protein from a PDB or CIF file.

\sphinxAtStartPar
This function reads a protein structure file (PDB or CIF) and extracts
the molecular hierarchy and residue information for simulation.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{combine\_residues\_to\_fragment() (src.Protein\_Class.ProteinStructure method)@\spxentry{combine\_residues\_to\_fragment()}\spxextra{src.Protein\_Class.ProteinStructure method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.combine_residues_to_fragment}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{combine\_residues\_to\_fragment}}}{\sphinxparam{\DUrole{n}{Temp\_Frag}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Combines a set of residues into a single fragment.

\sphinxAtStartPar
This method calculates the center of mass, new radius, and total mass
of a set of residues and groups them into a coarse\sphinxhyphen{}grained fragment.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{Temp\_Frag}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of residue data containing coordinates, radius, mass, and residue index.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_rigid\_body\_protein() (src.Protein\_Class.ProteinStructure method)@\spxentry{create\_rigid\_body\_protein()}\spxextra{src.Protein\_Class.ProteinStructure method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.create_rigid_body_protein}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{create\_rigid\_body\_protein}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Creates a rigid body representation of the protein.

\sphinxAtStartPar
This function sets up the protein as a rigid body for simulation, assigning
its molecular hierarchy and defining its diffusion properties.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_rigid\_body\_protein\_singlemodel() (src.Protein\_Class.ProteinStructure method)@\spxentry{create\_rigid\_body\_protein\_singlemodel()}\spxextra{src.Protein\_Class.ProteinStructure method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.create_rigid_body_protein_singlemodel}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{create\_rigid\_body\_protein\_singlemodel}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Creates a rigid body representation of a single\sphinxhyphen{}model protein.

\sphinxAtStartPar
This function loads the protein structure from a PDB or CIF file, sets up
its hierarchy, and defines a rigid body for coarse\sphinxhyphen{}grained simulation.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_current\_localization() (src.Protein\_Class.ProteinStructure method)@\spxentry{get\_current\_localization()}\spxextra{src.Protein\_Class.ProteinStructure method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.get_current_localization}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_current\_localization}}}{}{}
\pysigstopsignatures
\end{fulllineitems}

\index{setup\_residues() (src.Protein\_Class.ProteinStructure method)@\spxentry{setup\_residues()}\spxextra{src.Protein\_Class.ProteinStructure method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.setup_residues}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setup\_residues}}}{\sphinxparam{\DUrole{n}{mol\_id}}\sphinxparamcomma \sphinxparam{\DUrole{n}{residues}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Sets up and configures residues for the protein.

\sphinxAtStartPar
This method initializes residues, assigns attributes such as mass and diffusion,
and groups them into coarse\sphinxhyphen{}grained fragments based on the defined resolution.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{mol\_id}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Molecular identifier of the protein.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{residues}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of residue particles to be set up.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{translate\_protein\_to\_center() (src.Protein\_Class.ProteinStructure method)@\spxentry{translate\_protein\_to\_center()}\spxextra{src.Protein\_Class.ProteinStructure method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Protein_Class.ProteinStructure.translate_protein_to_center}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{translate\_protein\_to\_center}}}{\sphinxparam{\DUrole{n}{center}\DUrole{o}{=}\DUrole{default_value}{{[}0, 0, 0{]}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Translates the protein structure to a specified center position.

\sphinxAtStartPar
This function calculates the center of mass of the protein and applies
a transformation to shift the protein to the desired location.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{center}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} The target center coordinates. Default is {[}0, 0, 0{]}.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}



\subsection{src.Simulation\_Class module}
\label{\detokenize{src:module-src.Simulation_Class}}\label{\detokenize{src:src-simulation-class-module}}\index{module@\spxentry{module}!src.Simulation\_Class@\spxentry{src.Simulation\_Class}}\index{src.Simulation\_Class@\spxentry{src.Simulation\_Class}!module@\spxentry{module}}
\sphinxAtStartPar
Author: Neelesh Soni, \sphinxhref{mailto:neelesh@salilab.org}{neelesh@salilab.org}, \sphinxhref{mailto:neeleshsoni03@gmail.com}{neeleshsoni03@gmail.com}
Date: April 5, 2024
\index{logger (in module src.Simulation\_Class)@\spxentry{logger}\spxextra{in module src.Simulation\_Class}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.logger}}
\pysigstartsignatures
\pysigline{\sphinxcode{\sphinxupquote{src.Simulation\_Class.}}\sphinxbfcode{\sphinxupquote{logger}}}
\pysigstopsignatures
\sphinxAtStartPar
Description
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
TYPE

\end{description}\end{quote}

\end{fulllineitems}

\index{Simulation (class in src.Simulation\_Class)@\spxentry{Simulation}\spxextra{class in src.Simulation\_Class}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{ }}}\sphinxcode{\sphinxupquote{src.Simulation\_Class.}}\sphinxbfcode{\sphinxupquote{Simulation}}}{\sphinxparam{\DUrole{n}{param}}\sphinxparamcomma \sphinxparam{\DUrole{n}{system}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
A class to represent and execute Brownian dynamics simulations on a protein system.

\sphinxAtStartPar
The \sphinxtitleref{Simulation} class is responsible for setting up and running molecular simulations
using Brownian dynamics. It initializes the simulation environment, defines system constraints,
configures scoring functions, and records simulation trajectories.
\index{param (src.Simulation\_Class.Simulation attribute)@\spxentry{param}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.param}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{param}}}
\pysigstopsignatures
\sphinxAtStartPar
Dictionary containing simulation parameters.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
dict

\end{description}\end{quote}

\end{fulllineitems}

\index{system (src.Simulation\_Class.Simulation attribute)@\spxentry{system}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.system}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{system}}}
\pysigstopsignatures
\sphinxAtStartPar
The protein system being simulated.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
{\hyperref[\detokenize{src:src.System_Class.System}]{\sphinxcrossref{System}}}

\end{description}\end{quote}

\end{fulllineitems}

\index{model (src.Simulation\_Class.Simulation attribute)@\spxentry{model}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.model}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{model}}}
\pysigstopsignatures
\sphinxAtStartPar
The IMP model used for the simulation.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.Model

\end{description}\end{quote}

\end{fulllineitems}

\index{simulation\_time (src.Simulation\_Class.Simulation attribute)@\spxentry{simulation\_time}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.simulation_time}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{simulation\_time}}}
\pysigstopsignatures
\sphinxAtStartPar
Total duration of the simulation in seconds.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{temperature (src.Simulation\_Class.Simulation attribute)@\spxentry{temperature}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.temperature}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{temperature}}}
\pysigstopsignatures
\sphinxAtStartPar
Temperature of the simulation system.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{output\_dir (src.Simulation\_Class.Simulation attribute)@\spxentry{output\_dir}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.output_dir}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{output\_dir}}}
\pysigstopsignatures
\sphinxAtStartPar
Directory where simulation output files are stored.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
str

\end{description}\end{quote}

\end{fulllineitems}

\index{maximum\_move (src.Simulation\_Class.Simulation attribute)@\spxentry{maximum\_move}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.maximum_move}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{maximum\_move}}}
\pysigstopsignatures
\sphinxAtStartPar
Maximum displacement allowed per simulation step.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{bd\_step\_size\_sec (src.Simulation\_Class.Simulation attribute)@\spxentry{bd\_step\_size\_sec}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.bd_step_size_sec}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{bd\_step\_size\_sec}}}
\pysigstopsignatures
\sphinxAtStartPar
Step size for Brownian dynamics in seconds.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{output\_traj\_file (src.Simulation\_Class.Simulation attribute)@\spxentry{output\_traj\_file}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.output_traj_file}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{output\_traj\_file}}}
\pysigstopsignatures
\sphinxAtStartPar
Name of the trajectory output file.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
str

\end{description}\end{quote}

\end{fulllineitems}

\index{bd (src.Simulation\_Class.Simulation attribute)@\spxentry{bd}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.bd}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{bd}}}
\pysigstopsignatures
\sphinxAtStartPar
Brownian dynamics optimizer used for the simulation.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.atom.BrownianDynamics

\end{description}\end{quote}

\end{fulllineitems}

\index{scoring\_function (src.Simulation\_Class.Simulation attribute)@\spxentry{scoring\_function}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.scoring_function}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{scoring\_function}}}
\pysigstopsignatures
\sphinxAtStartPar
Scoring function used to evaluate simulation restraints.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.core.RestraintsScoringFunction

\end{description}\end{quote}

\end{fulllineitems}

\index{sim\_time\_ns (src.Simulation\_Class.Simulation attribute)@\spxentry{sim\_time\_ns}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.sim_time_ns}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{sim\_time\_ns}}}
\pysigstopsignatures
\sphinxAtStartPar
Total simulation time in nanoseconds.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{sim\_time\_frames (src.Simulation\_Class.Simulation attribute)@\spxentry{sim\_time\_frames}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.sim_time_frames}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{sim\_time\_frames}}}
\pysigstopsignatures
\sphinxAtStartPar
Number of simulation frames corresponding to the total simulation time.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
int

\end{description}\end{quote}

\end{fulllineitems}

\index{rmf\_dump\_interval\_frames (src.Simulation\_Class.Simulation attribute)@\spxentry{rmf\_dump\_interval\_frames}\spxextra{src.Simulation\_Class.Simulation attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.rmf_dump_interval_frames}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{rmf\_dump\_interval\_frames}}}
\pysigstopsignatures
\sphinxAtStartPar
Number of frames after which RMF trajectory snapshots are saved.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
int

\end{description}\end{quote}

\end{fulllineitems}

\index{convert\_time\_ns\_to\_frames() (src.Simulation\_Class.Simulation method)@\spxentry{convert\_time\_ns\_to\_frames()}\spxextra{src.Simulation\_Class.Simulation method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.convert_time_ns_to_frames}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{convert\_time\_ns\_to\_frames}}}{\sphinxparam{\DUrole{n}{time\_ns}}\sphinxparamcomma \sphinxparam{\DUrole{n}{step\_size\_fs}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Converts simulation time from nanoseconds to frames.

\sphinxAtStartPar
This function calculates the number of frames required based on the
specified simulation time and Brownian dynamics step size.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{time\_ns}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} Time in nanoseconds.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{step\_size\_fs}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} Step size in femtoseconds.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
Number of simulation frames.

\sphinxlineitem{Return type}
\sphinxAtStartPar
int

\end{description}\end{quote}

\end{fulllineitems}

\index{run() (src.Simulation\_Class.Simulation method)@\spxentry{run()}\spxextra{src.Simulation\_Class.Simulation method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.run}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{run}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Executes the simulation.

\sphinxAtStartPar
This function applies necessary updates to the system and runs the
Brownian dynamics optimizer for the specified number of simulation frames.
It logs the scoring function before and after the simulation.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{setup\_brownian\_dynamics() (src.Simulation\_Class.Simulation method)@\spxentry{setup\_brownian\_dynamics()}\spxextra{src.Simulation\_Class.Simulation method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Simulation_Class.Simulation.setup_brownian_dynamics}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{setup\_brownian\_dynamics}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Configures the Brownian dynamics simulation settings.

\sphinxAtStartPar
This function sets up simulation time parameters, scoring functions,
and RMF file configurations. It also defines constraints such as
the maximum movement and temperature.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}



\subsection{src.System\_Class module}
\label{\detokenize{src:module-src.System_Class}}\label{\detokenize{src:src-system-class-module}}\index{module@\spxentry{module}!src.System\_Class@\spxentry{src.System\_Class}}\index{src.System\_Class@\spxentry{src.System\_Class}!module@\spxentry{module}}
\sphinxAtStartPar
Author: Neelesh Soni
Contact: \sphinxhref{mailto:neelesh@salilab.org}{neelesh@salilab.org}, \sphinxhref{mailto:neeleshsoni03@gmail.com}{neeleshsoni03@gmail.com}
Date: Feb 5, 2024

\sphinxAtStartPar
Description:
This module defines the \sphinxtitleref{System} class, which represents the simulation system,
including proteins, interactions, and spatial constraints such as bounding boxes
and periodic boundary conditions (PBCs). The \sphinxtitleref{System} class integrates various
IMP (Integrative Modeling Platform) components to facilitate molecular simulations.
\begin{description}
\sphinxlineitem{Classes:}\begin{itemize}
\item {} 
\sphinxAtStartPar
System: Represents the molecular simulation system.

\end{itemize}

\end{description}
\index{logger (in module src.System\_Class)@\spxentry{logger}\spxextra{in module src.System\_Class}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.logger}}
\pysigstartsignatures
\pysigline{\sphinxcode{\sphinxupquote{src.System\_Class.}}\sphinxbfcode{\sphinxupquote{logger}}}
\pysigstopsignatures
\sphinxAtStartPar
Logger instance for logging system events.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
logging.Logger

\end{description}\end{quote}

\end{fulllineitems}

\index{System (class in src.System\_Class)@\spxentry{System}\spxextra{class in src.System\_Class}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{ }}}\sphinxcode{\sphinxupquote{src.System\_Class.}}\sphinxbfcode{\sphinxupquote{System}}}{\sphinxparam{\DUrole{n}{param}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
A class to represent the simulation system, including proteins, interactions, and spatial constraints.

\sphinxAtStartPar
This class provides methods for setting up molecular simulations using the Integrative Modeling Platform (IMP).
It defines proteins, interactions, and spatial boundaries within the system while applying restraints
such as excluded volume, Lennard\sphinxhyphen{}Jones potentials, and membrane exclusion constraints.
\index{bb (src.System\_Class.System attribute)@\spxentry{bb}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.bb}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{bb}}}
\pysigstopsignatures
\sphinxAtStartPar
Outer bounding box for the simulation.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.algebra.BoundingBox3D

\end{description}\end{quote}

\end{fulllineitems}

\index{bb\_cyt\_bbss (src.System\_Class.System attribute)@\spxentry{bb\_cyt\_bbss}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.bb_cyt_bbss}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{bb\_cyt\_bbss}}}
\pysigstopsignatures
\sphinxAtStartPar
Bounding box for the cytoplasm.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.core.BoundingBox3DSingletonScore

\end{description}\end{quote}

\end{fulllineitems}

\index{bb\_cyt\_harmonic (src.System\_Class.System attribute)@\spxentry{bb\_cyt\_harmonic}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.bb_cyt_harmonic}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{bb\_cyt\_harmonic}}}
\pysigstopsignatures
\sphinxAtStartPar
Harmonic upper bound for the cytoplasm.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.core.HarmonicUpperBound

\end{description}\end{quote}

\end{fulllineitems}

\index{bb\_cytoplasm (src.System\_Class.System attribute)@\spxentry{bb\_cytoplasm}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.bb_cytoplasm}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{bb\_cytoplasm}}}
\pysigstopsignatures
\sphinxAtStartPar
Bounding box for the cytoplasm.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.algebra.BoundingBox3D

\end{description}\end{quote}

\end{fulllineitems}

\index{bb\_harmonic (src.System\_Class.System attribute)@\spxentry{bb\_harmonic}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.bb_harmonic}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{bb\_harmonic}}}
\pysigstopsignatures
\sphinxAtStartPar
Harmonic upper bound for the simulation box.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.core.HarmonicUpperBound

\end{description}\end{quote}

\end{fulllineitems}

\index{bb\_nuc\_bbss (src.System\_Class.System attribute)@\spxentry{bb\_nuc\_bbss}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.bb_nuc_bbss}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{bb\_nuc\_bbss}}}
\pysigstopsignatures
\sphinxAtStartPar
Bounding box for the nucleus.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.core.BoundingBox3DSingletonScore

\end{description}\end{quote}

\end{fulllineitems}

\index{bb\_nuc\_harmonic (src.System\_Class.System attribute)@\spxentry{bb\_nuc\_harmonic}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.bb_nuc_harmonic}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{bb\_nuc\_harmonic}}}
\pysigstopsignatures
\sphinxAtStartPar
Harmonic upper bound for the nucleus.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.core.HarmonicUpperBound

\end{description}\end{quote}

\end{fulllineitems}

\index{bb\_nucleus (src.System\_Class.System attribute)@\spxentry{bb\_nucleus}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.bb_nucleus}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{bb\_nucleus}}}
\pysigstopsignatures
\sphinxAtStartPar
Bounding box for the nucleus.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.algebra.BoundingBox3D

\end{description}\end{quote}

\end{fulllineitems}

\index{h\_root (src.System\_Class.System attribute)@\spxentry{h\_root}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.h_root}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{h\_root}}}
\pysigstopsignatures
\sphinxAtStartPar
Root hierarchy of the system.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.atom.Hierarchy

\end{description}\end{quote}

\end{fulllineitems}

\index{interactions (src.System\_Class.System attribute)@\spxentry{interactions}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.interactions}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{interactions}}}
\pysigstopsignatures
\sphinxAtStartPar
List of interactions between proteins in the system.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{K\_BB (src.System\_Class.System attribute)@\spxentry{K\_BB}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.K_BB}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{K\_BB}}}
\pysigstopsignatures
\sphinxAtStartPar
Harmonic upper bound constant for bounding boxes.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{L (src.System\_Class.System attribute)@\spxentry{L}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.L}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{L}}}
\pysigstopsignatures
\sphinxAtStartPar
Length of the bounding box sides.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
int

\end{description}\end{quote}

\end{fulllineitems}

\index{mem\_thickness (src.System\_Class.System attribute)@\spxentry{mem\_thickness}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.mem_thickness}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{mem\_thickness}}}
\pysigstopsignatures
\sphinxAtStartPar
Thickness of the membrane.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
int

\end{description}\end{quote}

\end{fulllineitems}

\index{model (src.System\_Class.System attribute)@\spxentry{model}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.model}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{model}}}
\pysigstopsignatures
\sphinxAtStartPar
The IMP model for simulations.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.Model

\end{description}\end{quote}

\end{fulllineitems}

\index{outer\_bbss (src.System\_Class.System attribute)@\spxentry{outer\_bbss}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.outer_bbss}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{outer\_bbss}}}
\pysigstopsignatures
\sphinxAtStartPar
Outer bounding box singleton score.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.core.BoundingBox3DSingletonScore

\end{description}\end{quote}

\end{fulllineitems}

\index{proteins (src.System\_Class.System attribute)@\spxentry{proteins}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.proteins}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{proteins}}}
\pysigstopsignatures
\sphinxAtStartPar
List of proteins included in the system.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{restraints (src.System\_Class.System attribute)@\spxentry{restraints}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.restraints}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{restraints}}}
\pysigstopsignatures
\sphinxAtStartPar
List of applied restraints, including spatial and interaction constraints.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{state (src.System\_Class.System attribute)@\spxentry{state}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.state}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{state}}}
\pysigstopsignatures
\sphinxAtStartPar
Current state of the system.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.pmi.topology.State

\end{description}\end{quote}

\end{fulllineitems}

\index{sys (src.System\_Class.System attribute)@\spxentry{sys}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.sys}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{sys}}}
\pysigstopsignatures
\sphinxAtStartPar
IMP system topology object.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
IMP.pmi.topology.System

\end{description}\end{quote}

\end{fulllineitems}

\index{tor\_R (src.System\_Class.System attribute)@\spxentry{tor\_R}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.tor_R}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{tor\_R}}}
\pysigstopsignatures
\sphinxAtStartPar
Major radius for the toroidal membrane.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{tor\_r (src.System\_Class.System attribute)@\spxentry{tor\_r}\spxextra{src.System\_Class.System attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.tor_r}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{tor\_r}}}
\pysigstopsignatures
\sphinxAtStartPar
Minor radius for the toroidal membrane.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
float

\end{description}\end{quote}

\end{fulllineitems}

\index{Iterate\_Hierarchy() (src.System\_Class.System method)@\spxentry{Iterate\_Hierarchy()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.Iterate_Hierarchy}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{Iterate\_Hierarchy}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Iterates through the hierarchical structure of the system.

\sphinxAtStartPar
This function traverses the hierarchy of molecular components and logs
information about the system structure.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_LJ\_potential\_restraint() (src.System\_Class.System method)@\spxentry{add\_LJ\_potential\_restraint()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.add_LJ_potential_restraint}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_LJ\_potential\_restraint}}}{\sphinxparam{\DUrole{n}{param}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Adds a Lennard\sphinxhyphen{}Jones (LJ) potential restraint to the system.

\sphinxAtStartPar
This function applies an LJ potential to model attractive and repulsive interactions
between proteins in the system.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{param}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Dictionary containing LJ potential parameters, including:
\sphinxhyphen{} min\_dist (float): Minimum distance for interaction.
\sphinxhyphen{} max\_dist (float): Maximum distance for interaction.
\sphinxhyphen{} epsilon (float): Energy well depth.
\sphinxhyphen{} attractive\_weight (float): Weight for attractive forces.
\sphinxhyphen{} repulsive\_weight (float): Weight for repulsive forces.
\sphinxhyphen{} search\_dist\_cutoff (float): Cutoff distance for interaction search.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_components() (src.System\_Class.System method)@\spxentry{add\_components()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.add_components}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_components}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Adds protein components to the simulation based on the provided system parameters.
Proteins are created with specified copies, resolution, color, and diffusion coefficients.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
A list of Protein objects added to the system.

\sphinxlineitem{Return type}
\sphinxAtStartPar
list

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_excluded\_volume\_restraint() (src.System\_Class.System method)@\spxentry{add\_excluded\_volume\_restraint()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.add_excluded_volume_restraint}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_excluded\_volume\_restraint}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Adds an excluded volume restraint to prevent overlapping of proteins.

\sphinxAtStartPar
This function ensures that proteins are treated as spheres and prevents them from
overlapping by applying an excluded volume restraint using their constituent residues.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_interaction() (src.System\_Class.System method)@\spxentry{add\_interaction()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.add_interaction}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_interaction}}}{\sphinxparam{\DUrole{n}{protein1}}\sphinxparamcomma \sphinxparam{\DUrole{n}{protein2}}\sphinxparamcomma \sphinxparam{\DUrole{n}{interaction\_type}}\sphinxparamcomma \sphinxparam{\DUrole{n}{strength}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Creates and adds an interaction between two proteins in the system.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{protein1}} ({\hyperref[\detokenize{src:src.Protein_Class.Protein}]{\sphinxcrossref{\sphinxstyleliteralemphasis{\sphinxupquote{Protein}}}}}) \textendash{} The first protein involved in the interaction.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{protein2}} ({\hyperref[\detokenize{src:src.Protein_Class.Protein}]{\sphinxcrossref{\sphinxstyleliteralemphasis{\sphinxupquote{Protein}}}}}) \textendash{} The second protein involved in the interaction.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{interaction\_type}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Type of interaction (e.g., binding, electrostatic).

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{strength}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} Strength of the interaction.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_membrane\_exclusion\_restraint() (src.System\_Class.System method)@\spxentry{add\_membrane\_exclusion\_restraint()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.add_membrane_exclusion_restraint}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_membrane\_exclusion\_restraint}}}{\sphinxparam{\DUrole{n}{NGH\_proteins}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Adds a membrane exclusion restraint to prevent proteins from entering a defined membrane region.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{NGH\_proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of proteins near the membrane.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_membrane\_restraint() (src.System\_Class.System method)@\spxentry{add\_membrane\_restraint()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.add_membrane_restraint}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_membrane\_restraint}}}{\sphinxparam{\DUrole{n}{NGH\_proteins}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Adds a membrane restraint to proteins using a custom implementation.

\sphinxAtStartPar
This function applies a constraint that ensures proteins remain in a
specific membrane\sphinxhyphen{}associated region.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{NGH\_proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of proteins near the membrane.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_membrane\_restraint2() (src.System\_Class.System method)@\spxentry{add\_membrane\_restraint2()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.add_membrane_restraint2}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_membrane\_restraint2}}}{\sphinxparam{\DUrole{n}{NGH\_proteins}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Adds a membrane exclusion restraint using predefined NPC (Nuclear Pore Complex) methods.

\sphinxAtStartPar
This function applies a membrane restraint on selected proteins using a predefined
membrane exclusion model in IMP.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{NGH\_proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of proteins near the membrane.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{add\_restraint() (src.System\_Class.System method)@\spxentry{add\_restraint()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.add_restraint}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{add\_restraint}}}{\sphinxparam{\DUrole{n}{restraint}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Adds a restraint to the system to constrain molecular motion or interactions.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{restraint}} (\sphinxstyleliteralemphasis{\sphinxupquote{IMP.core.Restraint}}) \textendash{} The restraint object to be added.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{apply\_boundary\_conditions() (src.System\_Class.System method)@\spxentry{apply\_boundary\_conditions()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.apply_boundary_conditions}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{apply\_boundary\_conditions}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Applies periodic or spatial boundary conditions to the system by constraining
protein positions within predefined bounding regions.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{apply\_cytoplasm\_box\_boundary\_conditions() (src.System\_Class.System method)@\spxentry{apply\_cytoplasm\_box\_boundary\_conditions()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.apply_cytoplasm_box_boundary_conditions}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{apply\_cytoplasm\_box\_boundary\_conditions}}}{\sphinxparam{\DUrole{n}{proteins}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Applies bounding box constraints to the proteins inside the cytoplasm.

\sphinxAtStartPar
This function ensures that proteins within the cytoplasm remain within
the defined cytoplasmic bounding box using singleton restraints.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of Protein objects representing proteins in the cytoplasm.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{apply\_cytoplasm\_sphere\_boundary\_conditions() (src.System\_Class.System method)@\spxentry{apply\_cytoplasm\_sphere\_boundary\_conditions()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.apply_cytoplasm_sphere_boundary_conditions}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{apply\_cytoplasm\_sphere\_boundary\_conditions}}}{\sphinxparam{\DUrole{n}{proteins}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Applies spherical boundary conditions to the cytoplasm.

\sphinxAtStartPar
This function ensures that proteins inside the cytoplasm remain within the
defined cytoplasmic sphere boundary using singleton restraints.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of Protein objects representing proteins in the cytoplasm.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{apply\_nucleoplasm\_box\_boundary\_conditions() (src.System\_Class.System method)@\spxentry{apply\_nucleoplasm\_box\_boundary\_conditions()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.apply_nucleoplasm_box_boundary_conditions}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{apply\_nucleoplasm\_box\_boundary\_conditions}}}{\sphinxparam{\DUrole{n}{proteins}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Applies bounding box constraints to the proteins inside the nucleoplasm.

\sphinxAtStartPar
This function ensures that proteins within the nucleoplasm stay confined
within the defined nucleoplasmic bounding box using singleton restraints.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of Protein objects representing proteins in the nucleoplasm.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{apply\_nucleoplasm\_excluded\_boundary\_conditions() (src.System\_Class.System method)@\spxentry{apply\_nucleoplasm\_excluded\_boundary\_conditions()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.apply_nucleoplasm_excluded_boundary_conditions}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{apply\_nucleoplasm\_excluded\_boundary\_conditions}}}{\sphinxparam{\DUrole{n}{proteins}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Applies exclusion boundary conditions to the nucleoplasm.

\sphinxAtStartPar
This function ensures that proteins are excluded from the nucleoplasm by
enforcing a minimum distance constraint from the nucleoplasmic sphere.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of Protein objects representing proteins to be excluded from the nucleoplasm.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{apply\_nucleoplasm\_sphere\_boundary\_conditions() (src.System\_Class.System method)@\spxentry{apply\_nucleoplasm\_sphere\_boundary\_conditions()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.apply_nucleoplasm_sphere_boundary_conditions}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{apply\_nucleoplasm\_sphere\_boundary\_conditions}}}{\sphinxparam{\DUrole{n}{proteins}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Applies spherical boundary conditions to the nucleoplasm.

\sphinxAtStartPar
This function ensures that proteins inside the nucleoplasm remain within the
defined spherical boundary using singleton restraints.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of Protein objects representing proteins in the nucleoplasm.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_bounding\_box\_and\_pbc() (src.System\_Class.System method)@\spxentry{create\_bounding\_box\_and\_pbc()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.create_bounding_box_and_pbc}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{create\_bounding\_box\_and\_pbc}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Defines the bounding boxes and periodic boundary conditions (PBCs) for the simulation space.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_cell() (src.System\_Class.System method)@\spxentry{create\_cell()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.create_cell}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{create\_cell}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Creates a spherical representation of the cell with mass and color attributes.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
The created particle representing the cell.

\sphinxlineitem{Return type}
\sphinxAtStartPar
IMP.Particle

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_cell\_and\_nucleus\_bounding\_sphere() (src.System\_Class.System method)@\spxentry{create\_cell\_and\_nucleus\_bounding\_sphere()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.create_cell_and_nucleus_bounding_sphere}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{create\_cell\_and\_nucleus\_bounding\_sphere}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Creates bounding spheres for the cytoplasm and nucleus, defining the spatial constraints of the system.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_cell\_bounding\_sphere() (src.System\_Class.System method)@\spxentry{create\_cell\_bounding\_sphere()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.create_cell_bounding_sphere}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{create\_cell\_bounding\_sphere}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Creates a bounding sphere for the cytoplasm to define spatial constraints for cellular components.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{create\_nucleus() (src.System\_Class.System method)@\spxentry{create\_nucleus()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.create_nucleus}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{create\_nucleus}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Creates a spherical representation of the nucleus with mass and color attributes.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
The created particle representing the nucleus.

\sphinxlineitem{Return type}
\sphinxAtStartPar
IMP.Particle

\end{description}\end{quote}

\end{fulllineitems}

\index{shuffle\_neighborhood\_proteins() (src.System\_Class.System method)@\spxentry{shuffle\_neighborhood\_proteins()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.shuffle_neighborhood_proteins}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{shuffle\_neighborhood\_proteins}}}{\sphinxparam{\DUrole{n}{bounding\_box}}\sphinxparamcomma \sphinxparam{\DUrole{n}{NGH\_proteins}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Randomly shuffles the positions of proteins within a specified bounding box.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{bounding\_box}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Type of bounding box (‘nucleoplasm’, ‘cytoplasm’, or ‘default’).

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{NGH\_proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of proteins to shuffle.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{shuffle\_protein\_in\_box() (src.System\_Class.System method)@\spxentry{shuffle\_protein\_in\_box()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.shuffle_protein_in_box}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{shuffle\_protein\_in\_box}}}{\sphinxparam{\DUrole{n}{protein}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{shuffle\_protein\_in\_cell() (src.System\_Class.System method)@\spxentry{shuffle\_protein\_in\_cell()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.shuffle_protein_in_cell}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{shuffle\_protein\_in\_cell}}}{\sphinxparam{\DUrole{n}{protein}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{shuffle\_protein\_in\_cell\_nuc() (src.System\_Class.System method)@\spxentry{shuffle\_protein\_in\_cell\_nuc()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.shuffle_protein_in_cell_nuc}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{shuffle\_protein\_in\_cell\_nuc}}}{\sphinxparam{\DUrole{n}{protein}}}{}
\pysigstopsignatures
\end{fulllineitems}

\index{shuffle\_proteins() (src.System\_Class.System method)@\spxentry{shuffle\_proteins()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.shuffle_proteins}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{shuffle\_proteins}}}{\sphinxparam{\DUrole{n}{Proteins}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Randomly shuffles the positions of proteins within a specified bounding box.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{bounding\_box}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} The bounding box (‘nucleoplasm’, ‘cytoplasm’, ‘cell’, or ‘box’).

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{Proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of proteins to shuffle.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{update() (src.System\_Class.System method)@\spxentry{update()}\spxextra{src.System\_Class.System method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.System_Class.System.update}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{update}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Updates the interactions within the system.

\sphinxAtStartPar
This function iterates over all interactions in the system and applies the
specified interaction forces or constraints.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{None}}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}



\subsection{src.Analysis\_Class module}
\label{\detokenize{src:module-src.Analysis_Class}}\label{\detokenize{src:src-analysis-class-module}}\index{module@\spxentry{module}!src.Analysis\_Class@\spxentry{src.Analysis\_Class}}\index{src.Analysis\_Class@\spxentry{src.Analysis\_Class}!module@\spxentry{module}}\index{Analysis (class in src.Analysis\_Class)@\spxentry{Analysis}\spxextra{class in src.Analysis\_Class}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{ }}}\sphinxcode{\sphinxupquote{src.Analysis\_Class.}}\sphinxbfcode{\sphinxupquote{Analysis}}}{\sphinxparam{\DUrole{n}{trajectory\_file}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
A class for analyzing and visualizing protein trajectories from simulation data.

\sphinxAtStartPar
The \sphinxtitleref{Analysis} class provides methods to read simulation trajectory files, extract
protein movement data, convert trajectory data to JSON format, and generate visual
representations of protein movement.
\index{trajectory\_file (src.Analysis\_Class.Analysis attribute)@\spxentry{trajectory\_file}\spxextra{src.Analysis\_Class.Analysis attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.trajectory_file}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{trajectory\_file}}}
\pysigstopsignatures
\sphinxAtStartPar
Path to the RMF trajectory file.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
str

\end{description}\end{quote}

\end{fulllineitems}

\index{Protein\_Dict (src.Analysis\_Class.Analysis attribute)@\spxentry{Protein\_Dict}\spxextra{src.Analysis\_Class.Analysis attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.Protein_Dict}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{Protein\_Dict}}}
\pysigstopsignatures
\sphinxAtStartPar
Dictionary containing protein trajectory data,
where keys are protein names and values store radii and coordinate lists.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
OrderedDict

\end{description}\end{quote}

\end{fulllineitems}

\index{calculate\_interaction\_matrix() (src.Analysis\_Class.Analysis method)@\spxentry{calculate\_interaction\_matrix()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.calculate_interaction_matrix}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{calculate\_interaction\_matrix}}}{\sphinxparam{\DUrole{n}{trajectory\_data}}\sphinxparamcomma \sphinxparam{\DUrole{n}{distance\_threshold}\DUrole{o}{=}\DUrole{default_value}{30.0}}\sphinxparamcomma \sphinxparam{\DUrole{n}{selected\_proteins}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Computes an interaction matrix for protein\sphinxhyphen{}protein interactions based on their coordinates.
Merges different copies of the same protein type (e.g., NUP85\_0, NUP85\_1 \(\rightarrow\) NUP85).
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{trajectory\_data}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Dictionary containing protein trajectory data.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{distance\_threshold}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} Maximum distance (in Angstroms) for proteins to be considered interacting.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{selected\_proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} List of unique protein types (e.g., {[}‘NUP85’, ‘ULP1’{]}) to include.
If None, all proteins are included.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
Interaction matrix (NxN) where N is the number of unique protein types.
list: List of unique protein names.

\sphinxlineitem{Return type}
\sphinxAtStartPar
np.ndarray

\end{description}\end{quote}

\end{fulllineitems}

\index{calculate\_interaction\_matrix\_no\_grouping() (src.Analysis\_Class.Analysis method)@\spxentry{calculate\_interaction\_matrix\_no\_grouping()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.calculate_interaction_matrix_no_grouping}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{calculate\_interaction\_matrix\_no\_grouping}}}{\sphinxparam{\DUrole{n}{trajectory\_data}}\sphinxparamcomma \sphinxparam{\DUrole{n}{distance\_threshold}\DUrole{o}{=}\DUrole{default_value}{30.0}}\sphinxparamcomma \sphinxparam{\DUrole{n}{selected\_proteins}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Computes an interaction matrix for protein\sphinxhyphen{}protein interactions based on their coordinates.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{trajectory\_data}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Dictionary containing protein trajectory data.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{distance\_threshold}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} Maximum distance (in Angstroms) for proteins to be considered interacting.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{selected\_proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} List of protein names to include in the interaction matrix.
If None, all proteins are included.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
Interaction matrix (NxN) where N is the number of selected proteins.
list: List of selected protein names.

\sphinxlineitem{Return type}
\sphinxAtStartPar
np.ndarray

\end{description}\end{quote}

\end{fulllineitems}

\index{calculate\_msd() (src.Analysis\_Class.Analysis method)@\spxentry{calculate\_msd()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.calculate_msd}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{calculate\_msd}}}{\sphinxparam{\DUrole{n}{protein\_trajectory}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Calculates the Mean Square Displacement (MSD) for a given protein trajectory.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{protein\_trajectory}} (\sphinxstyleliteralemphasis{\sphinxupquote{np.ndarray}}) \textendash{} Nx3 array containing (x, y, z) coordinates over time.

\sphinxlineitem{Returns}
\sphinxAtStartPar
MSD values for different time lags.

\sphinxlineitem{Return type}
\sphinxAtStartPar
np.ndarray

\end{description}\end{quote}

\end{fulllineitems}

\index{calculate\_rdf() (src.Analysis\_Class.Analysis method)@\spxentry{calculate\_rdf()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.calculate_rdf}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{calculate\_rdf}}}{\sphinxparam{\DUrole{n}{trajectory\_data}}\sphinxparamcomma \sphinxparam{\DUrole{n}{selected\_proteins}}\sphinxparamcomma \sphinxparam{\DUrole{n}{bin\_size}\DUrole{o}{=}\DUrole{default_value}{5.0}}\sphinxparamcomma \sphinxparam{\DUrole{n}{max\_distance}\DUrole{o}{=}\DUrole{default_value}{100.0}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Computes the Radial Distribution Function (RDF) for selected proteins.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{trajectory\_data}} (\sphinxstyleliteralemphasis{\sphinxupquote{dict}}) \textendash{} Dictionary containing protein trajectory data.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{selected\_proteins}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of proteins for which RDF is computed.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{bin\_size}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} Bin width for RDF computation (in Angstroms).

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{max\_distance}} (\sphinxstyleliteralemphasis{\sphinxupquote{float}}) \textendash{} Maximum distance to consider (in Angstroms).

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
Distance bins.
np.ndarray: RDF values.

\sphinxlineitem{Return type}
\sphinxAtStartPar
np.ndarray

\end{description}\end{quote}

\end{fulllineitems}

\index{convert\_trajectory\_to\_json() (src.Analysis\_Class.Analysis method)@\spxentry{convert\_trajectory\_to\_json()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.convert_trajectory_to_json}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{convert\_trajectory\_to\_json}}}{\sphinxparam{\DUrole{n}{oname}}\sphinxparamcomma \sphinxparam{\DUrole{n}{precision}\DUrole{o}{=}\DUrole{default_value}{2}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Converts the trajectory data into a JSON file.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{oname}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the output JSON file.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{precision}} (\sphinxstyleliteralemphasis{\sphinxupquote{int}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Decimal precision for coordinates. Default is 2.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{normalize\_interaction\_matrix() (src.Analysis\_Class.Analysis method)@\spxentry{normalize\_interaction\_matrix()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.normalize_interaction_matrix}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{normalize\_interaction\_matrix}}}{\sphinxparam{\DUrole{n}{interaction\_matrix}}\sphinxparamcomma \sphinxparam{\DUrole{n}{method}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}global\textquotesingle{}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Normalizes the interaction matrix into probabilities.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{interaction\_matrix}} (\sphinxstyleliteralemphasis{\sphinxupquote{np.ndarray}}) \textendash{} Raw interaction frequency matrix.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{method}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Normalization method.
“row” (default) \(\rightarrow\) Row\sphinxhyphen{}wise normalization (conditional probabilities).
“global” \(\rightarrow\) Global normalization (all probabilities sum to 1).

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
Normalized probability matrix.

\sphinxlineitem{Return type}
\sphinxAtStartPar
np.ndarray

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_interaction\_matrix() (src.Analysis\_Class.Analysis method)@\spxentry{plot\_interaction\_matrix()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.plot_interaction_matrix}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{plot\_interaction\_matrix}}}{\sphinxparam{\DUrole{n}{interaction\_matrix}}\sphinxparamcomma \sphinxparam{\DUrole{n}{protein\_names}}\sphinxparamcomma \sphinxparam{\DUrole{n}{save\_path}\DUrole{o}{=}\DUrole{default_value}{None}}\sphinxparamcomma \sphinxparam{\DUrole{n}{title}\DUrole{o}{=}\DUrole{default_value}{\textquotesingle{}Interaction Matrix\textquotesingle{}}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Plots a heatmap of the normalized interaction matrix.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{interaction\_matrix}} (\sphinxstyleliteralemphasis{\sphinxupquote{np.ndarray}}) \textendash{} Normalized interaction matrix.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{protein\_names}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of protein names for axis labels.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{save\_path}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If provided, saves the plot instead of displaying.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_msd() (src.Analysis\_Class.Analysis method)@\spxentry{plot\_msd()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.plot_msd}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{plot\_msd}}}{\sphinxparam{\DUrole{n}{msd\_values}}\sphinxparamcomma \sphinxparam{\DUrole{n}{save\_path}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Plots the Mean Square Displacement (MSD) over time.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{msd\_values}} (\sphinxstyleliteralemphasis{\sphinxupquote{np.ndarray}}) \textendash{} Computed MSD values.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{save\_path}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If provided, saves the plot instead of displaying.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_protein\_trajectory() (src.Analysis\_Class.Analysis method)@\spxentry{plot\_protein\_trajectory()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.plot_protein_trajectory}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{plot\_protein\_trajectory}}}{\sphinxparam{\DUrole{n}{protein\_name}}\sphinxparamcomma \sphinxparam{\DUrole{n}{trajectory}}\sphinxparamcomma \sphinxparam{\DUrole{n}{save\_path}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Plots the 3D trajectory of a single protein.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{protein\_name}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the protein.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{trajectory}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of (x, y, z) coordinates for the protein trajectory.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{save\_path}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Path to save the plot. If None, the plot is displayed.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_rdf() (src.Analysis\_Class.Analysis method)@\spxentry{plot\_rdf()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.plot_rdf}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{plot\_rdf}}}{\sphinxparam{\DUrole{n}{r\_bins}}\sphinxparamcomma \sphinxparam{\DUrole{n}{rdf\_values}}\sphinxparamcomma \sphinxparam{\DUrole{n}{save\_path}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Plots the Radial Distribution Function (RDF).
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{r\_bins}} (\sphinxstyleliteralemphasis{\sphinxupquote{np.ndarray}}) \textendash{} Distance bins.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{rdf\_values}} (\sphinxstyleliteralemphasis{\sphinxupquote{np.ndarray}}) \textendash{} Computed RDF values.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{save\_path}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} If provided, saves the plot instead of displaying.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{plot\_two\_protein\_trajectory() (src.Analysis\_Class.Analysis method)@\spxentry{plot\_two\_protein\_trajectory()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.plot_two_protein_trajectory}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{plot\_two\_protein\_trajectory}}}{\sphinxparam{\DUrole{n}{protein\_name1}}\sphinxparamcomma \sphinxparam{\DUrole{n}{trajectory1}}\sphinxparamcomma \sphinxparam{\DUrole{n}{protein\_name2}}\sphinxparamcomma \sphinxparam{\DUrole{n}{trajectory2}}\sphinxparamcomma \sphinxparam{\DUrole{n}{save\_path}\DUrole{o}{=}\DUrole{default_value}{None}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Plots the 3D trajectories of two proteins.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}\begin{itemize}
\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{protein\_name1}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the first protein.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{trajectory1}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of (x, y, z) coordinates for the first protein.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{protein\_name2}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Name of the second protein.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{trajectory2}} (\sphinxstyleliteralemphasis{\sphinxupquote{list}}) \textendash{} List of (x, y, z) coordinates for the second protein.

\item {} 
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{save\_path}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}\sphinxstyleliteralemphasis{\sphinxupquote{, }}\sphinxstyleliteralemphasis{\sphinxupquote{optional}}) \textendash{} Path to save the plot. If None, the plot is displayed.

\end{itemize}

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{print\_json() (src.Analysis\_Class.Analysis method)@\spxentry{print\_json()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.print_json}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{print\_json}}}{\sphinxparam{\DUrole{n}{oname}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Loads and prints the JSON trajectory data.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{oname}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Path to the JSON file.

\sphinxlineitem{Returns}
\sphinxAtStartPar
None

\end{description}\end{quote}

\end{fulllineitems}

\index{read\_trajectory() (src.Analysis\_Class.Analysis method)@\spxentry{read\_trajectory()}\spxextra{src.Analysis\_Class.Analysis method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Analysis_Class.Analysis.read_trajectory}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{read\_trajectory}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Reads the simulation trajectory from an RMF file.

\sphinxAtStartPar
This function extracts protein coordinates at different frames and stores them in \sphinxtitleref{Protein\_Dict}.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
\begin{description}
\sphinxlineitem{A dictionary containing protein names as keys and a list of}
\sphinxAtStartPar
their radius and trajectory coordinates as values.

\end{description}


\sphinxlineitem{Return type}
\sphinxAtStartPar
OrderedDict

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}



\subsection{src.Input\_Parser module}
\label{\detokenize{src:module-src.Input_Parser}}\label{\detokenize{src:src-input-parser-module}}\index{module@\spxentry{module}!src.Input\_Parser@\spxentry{src.Input\_Parser}}\index{src.Input\_Parser@\spxentry{src.Input\_Parser}!module@\spxentry{module}}\index{InputParser (class in src.Input\_Parser)@\spxentry{InputParser}\spxextra{class in src.Input\_Parser}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Input_Parser.InputParser}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{class\DUrole{w}{ }}}\sphinxcode{\sphinxupquote{src.Input\_Parser.}}\sphinxbfcode{\sphinxupquote{InputParser}}}{\sphinxparam{\DUrole{n}{file\_path}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Bases: \sphinxcode{\sphinxupquote{object}}

\sphinxAtStartPar
A class for parsing and loading simulation parameters from a YAML configuration file.

\sphinxAtStartPar
The \sphinxtitleref{InputParser} class reads simulation parameters from a specified YAML file
and makes them available as a dictionary for use in the simulation setup.
\index{parameters (src.Input\_Parser.InputParser attribute)@\spxentry{parameters}\spxextra{src.Input\_Parser.InputParser attribute}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Input_Parser.InputParser.parameters}}
\pysigstartsignatures
\pysigline{\sphinxbfcode{\sphinxupquote{parameters}}}
\pysigstopsignatures
\sphinxAtStartPar
A dictionary containing the parsed simulation parameters.
\begin{quote}\begin{description}
\sphinxlineitem{Type}
\sphinxAtStartPar
dict

\end{description}\end{quote}

\end{fulllineitems}

\index{get\_parameters() (src.Input\_Parser.InputParser method)@\spxentry{get\_parameters()}\spxextra{src.Input\_Parser.InputParser method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Input_Parser.InputParser.get_parameters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{get\_parameters}}}{}{}
\pysigstopsignatures
\sphinxAtStartPar
Retrieves the simulation parameters.

\sphinxAtStartPar
This method returns the simulation parameters dictionary,
specifically extracting the “simulation\_parameters” section if available.
\begin{quote}\begin{description}
\sphinxlineitem{Returns}
\sphinxAtStartPar
The dictionary containing simulation parameters.

\sphinxlineitem{Return type}
\sphinxAtStartPar
dict

\end{description}\end{quote}

\end{fulllineitems}

\index{load\_simulation\_parameters() (src.Input\_Parser.InputParser method)@\spxentry{load\_simulation\_parameters()}\spxextra{src.Input\_Parser.InputParser method}}

\begin{fulllineitems}
\phantomsection\label{\detokenize{src:src.Input_Parser.InputParser.load_simulation_parameters}}
\pysigstartsignatures
\pysiglinewithargsret{\sphinxbfcode{\sphinxupquote{load\_simulation\_parameters}}}{\sphinxparam{\DUrole{n}{file\_path}}}{}
\pysigstopsignatures
\sphinxAtStartPar
Loads the simulation parameters from a YAML file.

\sphinxAtStartPar
This method reads the YAML file, parses its contents into a dictionary,
and stores it in the \sphinxtitleref{parameters} attribute.
\begin{quote}\begin{description}
\sphinxlineitem{Parameters}
\sphinxAtStartPar
\sphinxstyleliteralstrong{\sphinxupquote{file\_path}} (\sphinxstyleliteralemphasis{\sphinxupquote{str}}) \textendash{} Path to the YAML file containing simulation parameters.

\sphinxlineitem{Returns}
\sphinxAtStartPar
Dictionary containing the simulation parameters.

\sphinxlineitem{Return type}
\sphinxAtStartPar
dict

\end{description}\end{quote}

\end{fulllineitems}


\end{fulllineitems}



\chapter{Indices and tables}
\label{\detokenize{index:indices-and-tables}}\begin{itemize}
\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{genindex}

\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{modindex}

\item {} 
\sphinxAtStartPar
\DUrole{xref,std,std-ref}{search}

\end{itemize}


\renewcommand{\indexname}{Python Module Index}
\begin{sphinxtheindex}
\let\bigletter\sphinxstyleindexlettergroup
\bigletter{s}
\item\relax\sphinxstyleindexentry{src.Analysis\_Class}\sphinxstyleindexpageref{src:\detokenize{module-src.Analysis_Class}}
\item\relax\sphinxstyleindexentry{src.Input\_Parser}\sphinxstyleindexpageref{src:\detokenize{module-src.Input_Parser}}
\item\relax\sphinxstyleindexentry{src.Interaction\_Class}\sphinxstyleindexpageref{src:\detokenize{module-src.Interaction_Class}}
\item\relax\sphinxstyleindexentry{src.Protein\_Class}\sphinxstyleindexpageref{src:\detokenize{module-src.Protein_Class}}
\item\relax\sphinxstyleindexentry{src.Simulation\_Class}\sphinxstyleindexpageref{src:\detokenize{module-src.Simulation_Class}}
\item\relax\sphinxstyleindexentry{src.System\_Class}\sphinxstyleindexpageref{src:\detokenize{module-src.System_Class}}
\end{sphinxtheindex}

\renewcommand{\indexname}{Index}
\printindex
\end{document}